### PR TITLE
Cypher compiler unit tests

### PR DESCRIPTION
**Pull request by l-hoang**
Adds unit tests for the cypher compiler testing everything that we think should work up to this point in time. New tests should be added when new features are added.

### CODE REVIEW

File: tools/cypher-compile/test/cypher-compile-test.cpp [hunk 1]
```diff
@@ -0,0 +1,81 @@
+/**
+ * Copyright (C) 2020, KatanaGraph
+ */
+
+/**
+ * @file cypher-compile-test.cpp
+ *
+ * Various unit tests for the cypher compiler. Verifies that what it outputs is
+ * sane/expected. Note that there are some assumptions about the order the
+ * compiler parses things in present in these tests.
```
Comment (roshandathathri) [inline tools/cypher-compile/test/cypher-compile-test.cpp:10]: is present

File: tools/cypher-compile/test/cypher-compile-test.cpp [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+  std::string basic_node4 = "match path=(a:Test:Test2) return a;";
+  cc.compile(basic_node4.c_str());
+  verify_node(cc.getQueryNodes()[0],
+              galois::CompilerQueryNode{"0", "Test;Test2", "a", "path"});
+
+  // Print parsed values
+  galois::gInfo("Parsed single node values (not attached to an edge) "
+                "are as follows:");
+  for (const galois::CompilerQueryNode& a : cc.getQueryNodes()) {
+    a.printStruct(1);
```
Comment (roshandathathri) [inline tools/cypher-compile/test/cypher-compile-test.cpp:62]: It's better to avoid printing. This can be done later.

Comment (l-hoang) [reply tools/cypher-compile/test/cypher-compile-test.cpp:62]: yes; this is for writing tests since I see what the output of the compiler currently is

**Comment by l-hoang** [comment]
next few commits i'll push separately so you can see the difference in each commit @roshandathathri ; this one i forced pushed to account for the PR that just got merged

File: tools/cypher-compile/test/cypher-compile-test.cpp [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+
+int main() {
+  galois::SharedMemSys G;
+
+  galois::CypherCompiler cc;
+
+  // basic node testing first
+  std::string basic_node1 = "match (a) return a;";
+  cc.compile(basic_node1.c_str());
+  verify_node(cc.getQueryNodes()[0],
```
Comment (roshandathathri) [inline tools/cypher-compile/test/cypher-compile-test.cpp:60]: It would be useful to verify the number of query nodes and edges for all the tests.

Comment (l-hoang) [reply tools/cypher-compile/test/cypher-compile-test.cpp:60]: good point

**Comment by roshandathathri** [comment]
New changes LGTM

**Comment by l-hoang** [comment]
@roshandathathri see TODO comments for split edge test 2; what's the plan there/is this expected behavior?

**Comment by roshandathathri** [comment]
> for split edge test 2; what's the plan there/is this expected behavior?

Compiler should handle it in the future. For now, it's fine.

**Pull request merged**